src/haproxy.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+#
+# igcollect - haproxy
+#
+# Copyright (c) 2017 InnoGames GmbH
+#
+
+from argparse import ArgumentParser
+from subprocess import Popen, PIPE
+from time import time
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='haproxy')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    haproxy_info = Popen(
+        ('echo "show stat" | socat /var/run/haproxy.stat stdio'),
+        shell=True,
+        stdout=PIPE).stdout.read()
+    haproxy_info = haproxy_info.splitlines()
```

Comment:
It would have been more efficient to read lines one by one compared to reading them all and splitting.

Reply:
i am creating here a lines